Sources/SignalProducer.swift

```diff
@@ -1743,6 +1760,10 @@ extension SignalProducer {
 	/// - returns: A producer that sends events from `self` and then from
 	///            `replacement` when `self` completes.
 	public func then<U>(_ replacement: SignalProducer<U, Error>) -> SignalProducer<U, Error> {
+		return _then(replacement)
+	}
+
+	internal func _then<U>(_ replacement: SignalProducer<U, Error>) -> SignalProducer<U, Error> {
```

Comment:
Not having this leads to infinite self recursion, given that Swift selects the most specific overload for a given name.